src/Categories/Bicategory/Construction/Spans.agda

```diff
@@ -0,0 +1,406 @@
+{-# OPTIONS --without-K --safe #-}
+
+open import Categories.Category
+open import Categories.Diagram.Pullback
+
+module Categories.Bicategory.Construction.Spans {o ℓ e} (𝒞 : Category o ℓ e) (has-pullbacks : ∀ {X Y Z} (f : 𝒞 [ X , Z ]) (g : 𝒞 [ Y , Z ]) → Pullback 𝒞 f g) where
+
+open import Level
+open import Function using (_$_)
+
+open import Data.Product using (Σ; _×_; _,_; proj₁; proj₂; uncurry)
+
+open import Categories.Bicategory
+open import Categories.NaturalTransformation.NaturalIsomorphism hiding (refl; sym)
+
+open import Categories.Category.Construction.Spans 𝒞 renaming (Spans to Spans₁)
+
+open import Categories.Morphism
+open import Categories.Morphism.Reasoning 𝒞
+
+private
+  module Spans₁ X Y = Category (Spans₁ X Y)
+
+open Category 𝒞
+open HomReasoning
+open Equiv
+
+open Span
+open Span⇒
+open Pullback
+
+private
+  variable
+    A B C D E : Obj
+
+  --------------------------------------------------------------------------------
+  -- Various Spans, Span Maps, and their compositions
+  
+  id-span : Span A A
```

Comment:
Shouldn't the next 3 constructions be defined in `Categories.Category.Construction.Spans` ?  Actually, let me correct that: in `Categories.Category.Diagram.Span`.

Reply:
I could see `id-span` belonging there, but the other two constructions require the existence of pullbacks, so I think it may make more sense to have them live alongside the bicategory

Reply:
Well, you're right that this would all be conditional on some pullbacks existing - but that is part and parcel of spans composing, isn't it? So anyone wanting to compose spans will assume that those pullbacks exist. And then may end up rewriting the exact same functions.

In other words, I still believe they belong in `Categories.Category.Diagram.Span`, but in a sub-module with a `has-pullbacks` exactly like the one here.